semantic-analysis/src/Analysis/Syntax.hs

```diff
@@ -134,48 +106,66 @@ let' n v m = do
 
 -- Parsing
 
-parseFile :: Syntax rep => FilePath -> IO (Either (A.JSONPath, String) (Maybe rep))
+parseFile :: (Has (Throw String) sig m, MonadIO m) => FilePath -> m (File Term)
```

Comment:
We also parse into a `File`, which is convenient since we use those for analysis (so that we can service file/line/etc directives).